### PR TITLE
Support Webpack 5

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

Support Webpack 5. This PR adds support for webpack 5, but does not use it. I will post a webpack 5 validation pr next.

Validated in #10 

### Additional Details

@cerner/terra

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +43,9 @@
     "lodash.groupby": "^4.6.0",
     "semver": "^5.4.1"
   },
+  "peerDependencies": {
+    "webpack": "^4.30.0 || ^5.0.0"
```
Comment (mjhenkes) [inline package.json:47]: Normally it is non-passive to add a peer dependency, but this dependency was previously implied. How else are you going to use a webpack plugin? Now it's explicit.

File: lib/index.js [hunk 2]
```diff
@@ -96,10 +97,9 @@ DuplicatePackageCheckerPlugin.prototype.apply = function apply(compiler) {
       if (!isSeen) {
         const entry = { version, path: modulePath };
 
-        const issuer = module.issuer && module.issuer.resource
-          ? cleanPathRelativeToContext(module.issuer.resource)
-          : null;
-        entry.issuer = issuer;
+        const issuer = isWebpack5 ? compilation.moduleGraph.getIssuer(module) : module.issuer;
```
Comment (mjhenkes) [inline lib/index.js:100]: `module.issuer` works in webpack 5 but gives a deprecation notice.

File: lib/index.js [hunk 3]
```diff
@@ -96,10 +97,9 @@ DuplicatePackageCheckerPlugin.prototype.apply = function apply(compiler) {
...
+
+        entry.issuer = issuer ? cleanPathRelativeToContext(issuer.resource) : null;
```
Comment (emilyrohrbough) [inline lib/index.js]: Should this check for issuer.resource like you were previously?
```suggestion
        entry.issuer = issuer && issuer.resource ? cleanPathRelativeToContext(issuer.resource) : null;
```

Comment (mjhenkes) [reply lib/index.js]: Doesn't hurt to be safe.

**Review by ryanthemanuel — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Pull request merged**